bin/test-run

```diff
@@ -93,6 +93,11 @@ function helm_cleanup() {
     (
         set -e
         $helm_path --kube-context=$k8s_context --tiller-namespace=$tiller_namespace delete --purge $helm_release_name
+        # `helm delete` doesn't wait for resources to be deleted, so we wait explicitly.
+        # We wait for the namespace to be gone so the following call to `cleanup` doesn't fail when it attempts to delete
+        # the same namespace that is already being deleted here (error thrown by the NamespaceLifecycle controller).
+        # We don't have that problem with global resources, so no need to wait for them to be gone.
+        kubectl wait --for=delete ns/$linkerd_namespace-helm --timeout=40s
```

Comment:
should this be `ns/$tiller_namespace`? doesn't `$linkerd_namespace-helm` get cleaned up in `bin/test-cleanup`?

Reply:
The `helm delete` command right before this starts the deletion of Linkerd whose namespace is indeed `$linkerd_namespace-helm`, and we wanna wait till that is gone.
The line immediately below calls `helm reset` which does remove Tiller at the `$tiller_namespace` namespace.
We could certainly have `bin/test-cleanup` take care of that, but the intention here was to test that `helm delete` and `helm reset` are working as expected.

Reply:
Thanks this is helpful. I didn't understand the relationship between `helm delete` vs. `helm reset` and `$linkerd_namespace-helm` vs. `$tiller_namespace`.

Mind adding a couple comments around this:

```diff
diff --git a/bin/test-run b/bin/test-run
index 18db3a0e..07b8dfb0 100755
--- a/bin/test-run
+++ b/bin/test-run
@@ -92,12 +92,14 @@ function run_helm_test() {
 function helm_cleanup() {
     (
         set -e
+        # `helm delete` deletes $linkerd_namespace-helm
         $helm_path --kube-context=$k8s_context --tiller-namespace=$tiller_namespace delete --purge $helm_release_name
         # `helm delete` doesn't wait for resources to be deleted, so we wait explicitly.
         # We wait for the namespace to be gone so the following call to `cleanup` doesn't fail when it attempts to delete
         # the same namespace that is already being deleted here (error thrown by the NamespaceLifecycle controller).
         # We don't have that problem with global resources, so no need to wait for them to be gone.
         kubectl wait --for=delete ns/$linkerd_namespace-helm --timeout=40s
+        # `helm reset` deletes the tiller pod in $tiller_namespace
         $helm_path --kube-context=$k8s_context --tiller-namespace=$tiller_namespace reset
         kubectl --context=$k8s_context delete clusterrolebinding ${tiller_namespace}:tiller-cluster-admin
         echo $tiller_namespace
```